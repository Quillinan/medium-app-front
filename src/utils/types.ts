export interface ErrorResponse {
  code: string;
  message: string;
  detailedMessage: string;
  helpUrl: string;
  details: null;
}

export interface Birthday {
  NOME: string;
  DTNASCIMENTO: number;
}
