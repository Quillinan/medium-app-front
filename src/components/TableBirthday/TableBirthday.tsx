import React from 'react';
import { TableBirthdayProps } from '@utils/Types/Types';

const TableBirthday: React.FC<TableBirthdayProps> = ({ data }) => {
  return (
    <div className='grid grid-cols-1 sm:grid-cols-2 gap-4'>
      {data.length > 0 ? (
        data.map((person, index) => (
          <div
            key={index}
            className='bg-white border border-gray-400 rounded-lg shadow-md p-4 flex items-center space-x-4'
          >
            {/* Renderizar a imagem se ela existir */}
            {person.IMAGEM && (
              <img
                src={`data:image/jpeg;base64,${person.IMAGEM}`}
                alt={`Foto de ${person.NOME}`}
                className='w-16 h-16 rounded-full object-cover'
              />
            )}
            <div>
              <h3 className='text-lg font-semibold text-gray-800'>
                {person.NOME}
              </h3>
              <p className='text-sm text-gray-600'>Setor: {person.DESCRICAO}</p>
              <p className='text-sm text-gray-600'>
                Anivers√°rio: {person.DTNASCIMENTO}
              </p>
            </div>
          </div>
        ))
      ) : (
        <p className='text-center text-sm text-gray-500 col-span-full'>
          Nenhum aniversariante encontrado.
        </p>
      )}
    </div>
  );
};

export default TableBirthday;
