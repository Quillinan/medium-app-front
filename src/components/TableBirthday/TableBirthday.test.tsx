import { TableBirthdayProps } from '@utils/types';
import TableBirthday from './TableBirthday';
import { render, screen } from '@testing-library/react';

describe('TableBirthday', () => {
  const mockData: TableBirthdayProps['data'] = [
    { NOME: 'João', DTNASCIMENTO: 1 },
    { NOME: 'Maria', DTNASCIMENTO: 17 },
  ];
  it('should render a message when no data is provided', () => {
    render(<TableBirthday data={[]} />);

    const noDataMessage = screen.getByText(
      /Nenhum aniversariante encontrado./i
    );
    expect(noDataMessage).toBeInTheDocument;
  });
  it('should render the table header correctly', () => {
    render(<TableBirthday data={[]} />);

    const nomeHeader = screen.getByText(/Nome/i);
    const diaHeader = screen.getByText(/Dia/i);

    expect(nomeHeader).toBeInTheDocument;
    expect(diaHeader).toBeInTheDocument;
  });

  it('should render the data rows when data is provided', () => {
    render(<TableBirthday data={mockData} />);

    expect(screen.getByText('João')).toBeInTheDocument;
    expect(screen.getByText('Maria')).toBeInTheDocument;
    expect(screen.getByText('1')).toBeInTheDocument;
    expect(screen.getByText('17')).toBeInTheDocument;
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<TableBirthday data={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
