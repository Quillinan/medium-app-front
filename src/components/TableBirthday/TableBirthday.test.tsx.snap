// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableBirthday > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-x-auto"
  >
    <table
      class="min-w-full bg-white border border-gray-400 rounded-lg shadow-md"
    >
      <thead>
        <tr>
          <th
            class="px-4 py-2 border-b-2 border-gray-400 bg-gray-200 text-left text-sm font-semibold text-gray-700 uppercase tracking-wider"
          >
            Nome
          </th>
          <th
            class="px-4 py-2 border-b-2 border-gray-400 bg-gray-200 text-center text-sm font-semibold text-gray-700 uppercase tracking-wider"
          >
            Dia
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="px-4 py-2 text-center text-sm text-gray-500"
            colspan="2"
          >
            Nenhum aniversariante encontrado.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
